syntax = "proto3";

service GameMaster {
  rpc SetSymbol(SetSymbolRequest) returns (SetSymbolResponse) {}
  rpc ListBoard(ListBoardRequest) returns (ListBoardResponse) {}
}

message SetSymbolRequest {
  int32 node_id = 1;
  int32 position = 2;
}

message SetSymbolResponse {
  bool success = 1;
  string error = 2;
}

message ListBoardRequest {}

message ListBoardResponse {
  repeated int32 board = 1;
  string move_timestamps = 2;
}
